### ISSUE TITLE
Add more integration tests to check Diktat compatibility with KtLint

### ISSUE THREAD

**Issue by 0x6675636b796f75676974687562** [cqfn/diKTat#1569]
**Update:** we already have _Gradle_ plug-in tests available under `diktat-gradle-plugin/src/functionalTest/`,
 - broken with #1428,
 - re-enabled with #1579,

so the primary focus of this task is the **Spotless plug-in** (see below).

# Diktat _Gradle_ plug-in

Probably `DiktatJavaExecTaskTest`, feel free to suggest a different test class name.

1. Check that the plug-in is indeed executed and reports errors (i. e. `./gradlew diktatCheck`) against invalid input should produce errors from `diktat-ruleset` namespace.
   * `plain` reporter
   * `sarif` reporter
2. Check that `KtLint`-standard rules are indeed **excluded**, i. e. none of the errors from the `standard` or `experimental` namespaces are triggered by the invalid input (be sure to also disable the corresponding Diktat rules via `diktat-analysis.yml`). This test should fail if changes from #1568 are temporarily rolled back.

# Spotless _Gradle_ plug-in

1. Check that [diffplug/spotless](https://github.com/diffplug/spotless), when run as a [_Gradle_ plug-in](https://github.com/diffplug/spotless/tree/main/plugin-gradle#diktat) with the **snapshot** Diktat version, produces expected errors and has no regressions like #1559.
   * `plain` reporter
   * `sarif` reporter

Basically, we need a test which would, one way or another, launch a forked JVM against a test project, something we do in `DiktatSmokeTest`.

At the same time, I'd vote against using `save-cli` for this purpose and instead look for smth similar to [khmarbaise/maven-it-extension](https://github.com/khmarbaise/maven-it-extension) for JUnit 5, for two reasons:

 - `save-cli` takes a long time to download over a slow network,
 - a forked JVM is much easier to debug in case of test failures.

@petertrr, @nulls, WDYT?

**Comment by nulls** [comment]
we need also test spotless maven plugin

**Comment by petertrr** [comment]
Sounds reasonable; our `functionalTest` for Gradle plugin can be enhanced to run different scenarios as well as projects with `spotless` plugin using Gradle testkit.

**Comment by nulls** [comment]
Depends on https://github.com/saveourtool/save-cli/issues/521
